runtime/src/macros.rs

```diff
@@ -0,0 +1,40 @@
+//! Macros shared across runtime implementations.
+
+/// Prepare metrics for a spawned task.
+#[macro_export]
+macro_rules! spawn_metrics {
+    ($ctx:ident, future) => {{
+        let label = $crate::telemetry::metrics::task::Label::future($ctx.name.clone());
+        $ctx.executor
+            .metrics
+            .tasks_spawned
+            .get_or_create(&label)
+            .inc();
+        let gauge = $ctx
+            .executor
+            .metrics
+            .tasks_running
+            .get_or_create(&label)
+            .clone();
+        (label, gauge)
+    }};
+    ($ctx:ident, blocking, $dedicated:expr) => {{
+        let label = if $dedicated {
+            $crate::telemetry::metrics::task::Label::blocking_dedicated($ctx.name.clone())
+        } else {
+            $crate::telemetry::metrics::task::Label::blocking_shared($ctx.name.clone())
+        };
+        $ctx.executor
+            .metrics
+            .tasks_spawned
+            .get_or_create(&label)
+            .inc();
+        let gauge = $ctx
+            .executor
+            .metrics
+            .tasks_running
+            .get_or_create(&label)
+            .clone();
+        (label, gauge)
+    }};
+}
```

Comment:
Not super familiar with macros but is it standard to have this `future` vs `blocking` in the arguments, versus just having two differently-named macros? I feel like this way is just more confusing

Reply:
IMHO macro use is a free-for-all of style (which makes them both very powerful but also very dangerous).

I cleaned up the one here to share the metrics logic (and now think it meets the hurdle for doing it this way).